pkg/blob/controllerserver.go

```diff
@@ -417,7 +424,14 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 				return nil, status.Errorf(codes.Internal, "failed to GetStorageAccesskey on account(%s) rg(%s), error: %v", accountOptions.Name, accountOptions.ResourceGroup, err)
 			}
 		}
-		if err := d.copyVolume(ctx, req, accountKey, validContainerName, storageEndpointSuffix); err != nil {
+		var accountSasToken string
```

Comment:
write like this:
```
useAccountSASToken := len(secrets) > 0
d.copyVolume(ctx, req, accountKey, validContainerName, storageEndpointSuffix, useAccountSASToken)
```

if secrets are provided, it could be account key provided, then the driver should also get sas token though accountSasToken is not provided.